handlers/serviceset.go

```diff
@@ -176,19 +181,57 @@ func (s *ServiceSet) logError(format string, args ...interface{}) {
 	}
 }
 
-// reloadEndpointHandler
-func (s *ServiceSet) reloadEndpointHandler(w http.ResponseWriter, r *http.Request) {
+func (s *ServiceSet) loadEndpointHandler(w http.ResponseWriter, r *http.Request) {
 	token := r.URL.Query().Get("token")
-	if !s.enableReloadSignal || s.reloadToken == "" {
+	path := r.URL.Query().Get("path")
+	if s.authToken == "" || path == "" {
 		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
 		return
-	} else if s.enableReloadSignal && token == s.reloadToken {
+	} else if token == s.authToken {
 		_, err := w.Write([]byte("OK"))
 		if err != nil {
 			s.logError("Error rendering response during reload: %v", err)
 			return
 		}
-		syscall.Kill(syscall.Getpid(), syscall.SIGHUP)
+		id, err := s.generateID(path, s.baseDir)
+		if err != nil {
+			s.logError("Could not create ID for tileset %q\n%v", path, err)
+			return
+		}
+		err = s.AddTileset(path, id)
```

Comment:
Oh, it's nice that these endpoints are mainly wrappers to an existing method :heart: